backend/lib/models/User.js

```diff
@@ -1,33 +1,50 @@
-const { model, Schema } = require("mongoose");
+const { Schema, model } = require("mongoose");
+const { schema: locationSchema } = require("./Location");
+const { model: Post } = require("./Post");
+const { model: Comment } = require("./Comment");
+const { isValidEmail } = require("../utils");
 
-const UserSchema = new Schema(
+const userSchema = new Schema(
   {
-    _id: {
-      required: true,
-      type: Schema.Types.ObjectId,
-    },
-    address: {
-      type: String,
-    },
-    location: Object,
-    name: {
+    about: { maxLength: 100, trim: true, type: String },
+    authId: { required: true, type: String },
+    email: {
       required: true,
       type: String,
+      validator: isValidEmail,
     },
-    needs: {
-      type: Array,
-    },
-    type: String,
-    wants: {
-      type: Array,
-    },
-  },
-  {
-    timestamps: true,
+    location: locationSchema,
+    photo: String,
   },
+  { collection: "users", timestamps: true },
 );
 
-const User = model("User", UserSchema);
+function updateAuthorLocationReference(location) {
+  Post.where(
+    { "author.authorId": this._id },
+    { $set: { "author.location": location } },
+  );
+  Comment.where(
+    { "author.authorId": this._id },
+    { $set: { "author.location": location } },
+  );
+
+  return location;
+}
+
+userSchema.path("location", {
+  set: updateAuthorLocationReference,
+});
```

Comment:
Actually, we don't need to keep the location ref synced.
If someone updates their location, the post/comments shouldn't move with them, so we don't need this callback here.